packages/rocket-auth-aws-infrastructure/src/lambdas/package.json

```diff
@@ -0,0 +1,18 @@
+{
+  "name": "lambdas",
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.ts",
+  "author": "",
+  "license": "ISC",
+  "devDependencies": {
+    "validator": "^13.1.1",
+    "@boostercloud/framework-types": "0.9.4",
+    "tslib": "2.0.1"
+  },
+  "bundledDependencies": [
+    "validator",
+    "@boostercloud/framework-types",
+    "tslib"
+  ]
+}
```

Comment:
@javiertoledo Totally agreed, but for some reason, I had to upload the `dist` folder with all lambdas dependencies in order to work. I didn't find a better way but definitely, it's something we need to discuss because this was one of the pain points I found creating this rocket.

Reply:
One idea we discussed is that rockets like this are provided in two (or more) separate packages (See the [naming recommendation section in the docs](https://github.com/boostercloud/booster/tree/master/docs#naming-recommendations)):

In this case:

* `rocket-auth-aws-infrastructure`: Provides all the code that is imported dynamically from the CLI to provision and deploy the infrastructure.
* `rocket-auth-aws`: Provides extensions to the Booster app, so you could provide the lambdas as functions. The only problem is that you'd need to re-export them in the App index doing something like this:

```typescript
// index.ts from a Booster application
import { Booster } from '@boostercloud/framework-core'
export {
  Booster,
  boosterEventDispatcher,
  boosterPreSignUpChecker,
  boosterServeGraphQL,
  boosterNotifySubscribers,
  boosterTriggerScheduledCommand,
} from '@boostercloud/framework-core'

// Re-export the functions provided by the rocket
export * from '@boostercloud/rocket-auth-aws'

Booster.start(__dirname)
```

Making the end user re-export the lambdas provided by the rockets is a bit messy, I wonder id we could do something more "standarized", like having a lambda registry object or something like that. We're already exporting the `Booster` object, which is a proxy for the `BoosterConfig` object. If we register all lambdas in a new `lambda` property, I think we could change the Booster template to export only the `Booster` object like this:

```typescript
import { Booster } from '@boostercloud/framework-core'
export { Booster } from '@boostercloud/framework-core'

Booster.start(__dirname)
```

And then from the infrastructure packages, referencing the functions as `Booster.lambda.boosterServeGraphQL` instead of just `boosterServeGraphQL`. The advantage of this, if it works, is that rockets could register any number of lambdas they want in an standardized way without hacks, and as a side effect, we could get rid of the weird static functions in the `framework-core` package (https://github.com/boostercloud/booster/blob/master/packages/framework-core/src/booster.ts#L125-L143)

What do you think @boostercloud/booster-core?

Reply:
Love the proposal! But sadly what you show here, won't work:

```ts
import { Booster } from '@boostercloud/framework-core'
export { Booster } from '@boostercloud/framework-core'

Booster.start(__dirname)
```

As AWS requires you to have static functions in the global scope in order to call them from lambdas :(((((((

Reply:
I see a potential issue here, because in this paeticular case `auth-aws` needs to interact with `auth-aws-infrastructure` because the infrastructure needs to create thec ambda itself in the stack and the package needs the lambda code available as a folder or a zip file. Am I missing something?

Reply:
@NickSeagull could we inject them in the global scope as they're "registered" somehow?

```typescript
// Would this even work??
global['nameOfTheLambda'] = leFunction
```

Reply:
@javiertoledo We investigated a lot this matter but never thought of that approach 😳 

If it works, it would be amazing!